pkg/scheduler/controller/shoot/reconciler.go

```diff
@@ -142,6 +140,10 @@ func determineSeed(
 	if err != nil {
 		return nil, err
 	}
+	filteredSeeds, err = filterHighAvailabilitySeeds(filteredSeeds, shoot)
```

Comment:
As discussed, please also adjust `docs/concepts/scheduler.md` to the changes here.